src/GraphNodeManagerViewExtension/GraphNodeManagerViewModel.cs

```diff
@@ -325,18 +325,37 @@ internal void ExportGraph(object parameter)
         {
             if (parameter == null) return;
             string type = parameter.ToString();
+            string promptName =  System.IO.Path.GetFileNameWithoutExtension(currentWorkspace.FileName);
+
+            var filteredNodes = FilteredNodesArray();
 
             switch (type)
             {
                 case "CSV":
-                    Utilities.Utilities.ExportToCSV(Nodes.ToArray());
+                    Utilities.Utilities.ExportToCSV(filteredNodes, promptName);
                     break;
                 case "JSON":
-                    Utilities.Utilities.ExportToJson(Nodes.ToArray());
+                    Utilities.Utilities.ExportToJson(filteredNodes, promptName);
                     break;
             }
         }
 
+        /// <summary>
+        /// Helper method to return an Array of the currently active Nodes
+        /// </summary>
+        /// <returns></returns>
+        private GridNodeViewModel [] FilteredNodesArray()
+        {
+            var filteredNodes = new List<GridNodeViewModel>();
```

Comment:
if this is called frequently or there are a large number of these nodes you can either directly allocate an array with the Enumerable's count to skip allocating a list, or you can set the expected length of the list so it does not need to keep increasing capacity as you add - since you already know how big the itemsSource is.

Reply:
Of course, you are right. It is a very low-volume method, though, and ItemSource is an IEnumerable, which doesn't seem to have a 'Count' property. But I changed that to 
`return GraphNodeManagerView.NodesInfoDataGrid.ItemsSource.Cast<GridNodeViewModel>().ToArray();`
Would that work?